index.bs

```diff
@@ -1598,286 +1600,161 @@ A URL for Blob and File reference</h2>
+add an entry to the Blob URL Store</dfn> for a given |object|,
+run the following steps:
 
-<h4 id="originOfBlobURL">
-Origin of Blob URLs</h4>
+1. Let |store| be the user agent's [=Blob URL Store=];
+1. Let |url| be the result of [=generating a new Blob URL=].
+1. Let |entry| be a new [=Blob URL Entry=] consisting of |object| and the [=current settings object=].
+1. [=map/Set=] |store|[|url|] to |entry|.
+1. Return |url|.
 
-<a>Blob URLs</a> are created using <code>URL.{{URL/createObjectURL()}}</code>,
-and are revoked using <code>URL.{{URL/revokeObjectURL()}}</code>.
-The <dfn lt="origin of a Blob URL|Blob URL's origin">origin of a Blob URL</dfn> must be the same as the [=environment settings object/origin=]
-specified by the <a>current settings object</a>
-at the time the method that created it was called.
+</div>
 
-Issue: there is currently some confusion between the generic definition of the
-[=url/origin|origin of a URL=] and the specific definition of the [=origin of a Blob URL=]. This is
-tracked in <a href="https://github.com/w3c/FileAPI/issues/63">issue #63</a> and in
-<a href="https://github.com/whatwg/url/issues/127">whatwg/url#127</a>.
+<div algorithm="removeBlobURL">
+To <dfn id="removeTheEntry" lt="remove an entry|remove the entry|remove an entry from the Blob URL Store|remove the entry from the Blob URL Store" for="blob url store">
+remove an entry from the Blob URL Store</dfn> for a given |url|,
+run the following steps:
 
-[=CORS protocol|Cross-origin requests=] on Blob URLs must return a <a>network error</a>.
+1. Let |store| be the user agent's [=Blob URL Store=];
+1. Let |url string| be the result of [=URL serializer|serializing=] |url|.
+1. Remove |store|[|url string|].
 
-Note: In practice this means that HTTP and HTTPS [=/origins=]
-are covered by this specification as valid origins for use with Blob URLs.
-This specification does not address the case of non-HTTP and non-HTTPS [=/origins=].
-For instance <code>blob:file:///Users/arunranga/702efefb-c234-4988-a73b-6104f1b615ee</code>
-(which uses the "file:" [=/origin=])
-may have behavior that is undefined,
-even though user agents may treat such Blob URLs as valid.
+Issue: Should this ignore fragments, queries and/or extra path components?
 
-<h4 id="unicodeSerializationOfBlobURL">
-Serialization of a Blob URL</h4>
+</div>
 
-The <dfn id="unicodeBlobURL">Serialization of a Blob URL</dfn>
-is the value returned by the following algorithm,
-which is invoked by <code>URL.{{URL/createObjectURL()}}</code>:
+<div algorithm="createBlobURL">
+To <dfn id="unicodeBlobURL" lt="generate a new Blob URL|generating a new Blob URL">
+generate a new Blob URL</dfn>, run the following steps:
 
 1. Let |result| be the empty string.
-  Append the string "blob"
-  (that is, the Unicode code point sequence U+0062, U+006C, U+006F, U+0062)
+1. Append the string "blob"
+  (that is, the Unicode [=code point=] sequence U+0062, U+006C, U+006F, U+0062)
```

Comment:
You could even make this append the string "blob:" since strings are a well-defined concept now. I wonder though if this is the way we should generate them or if we should create a URL record and then serialize that.